design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,330 @@
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to enter the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user skips this step using the skip button.
+6. The wizard sends a GET request to the templating service with a query string of the format:
+    ```
+    ?template_location=databricksConfig.props.tt
+    ```
+7. The templating service will fail to satisfy the `lakefs_url` template property and will return **‘400 Bad Request: error code 1’** to the wizard.
+8. The wizard will prompt a message saying that some needed information were not specified and that he should make sure he typed everything along the way.
+9. Continue with the flow as described above…
+
+### Sad flow - No fetching permissions
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to enter the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user skips this step using the skip button.
+6. The wizard sends a GET request to the templating service with a query string of the format:
+    ```
+    ?lakefs_url=<url>&template_location=databricksConfig.props.tt
+    ```
+7. The templating service tries to fetch the template from the provided location and will fail to do so as the server doesn’t have sufficient permissions.
+8. The templating service will return **‘500 Internal Server Error’** to the wizard.
+9. The wizard will prompt a message saying that the server could not access the requested template.
+10. Continue with the flow as described above…
+
+---
+
+## Monitoring
+
+### Operative Metrics
+
+1. Templating service was called
+    ```json
+    {
+        "class": "templating_service",
+        "name": "calling_service",
+        "value": "<service name>"
+    }
+    ```
+2. Templating service- status 200
+    ```json
+    {
+        "class": "templating_service",
+        "name": "successful_call",
+        "value": "<service name>"
+    }
+    ```
+3. Templating service - status 500 - no access to provided template location
+    ```json
+    {
+        "class": "templating_service",
+        "name": "no_access",
+        "value": "<service name>"
+    }
+    ```
+4. Templating service - status 5xx - general
+    ```json
+    {
+        "class": "templating_service",
+        "name": "5xx",
+        "value": "<service name>"
+    }
+    ```
+5. Templating service - status 4xx - general
+    ```json
+    {
+        "class": "templating_service",
+        "name": "4xx",
+        "value": "<service name>"
+    }
+    ```
+
+### BI Metrics
+
+Sent directly from the GUI Wizard
```

Comment:
I think that all of these should identify which wizard it was.  Initially we expect to have only one, but later we expect to have more!

Similarly, we might report which Spark configs were generated.

Reply:
They identify it as `spark_wizard`, did you mean something else?